### PR TITLE
Move batch restrictions from HTTP to the main spec

### PR DESCRIPTION
**Pull request by jskeet**
The restriction in terms of uniqueness is new, and definitely open for debate. I had previously always just assumed it. If we choose not to require it, we should explicitly permit duplicates.

### CODE REVIEW

File: cloudevents/spec.md [hunk 1]
```diff
@@ -148,6 +148,15 @@ as one that is unaware of CloudEvents.
...
 encoded in a single message body, according to a specific event format. Not
 all event formats or protocol bindings support batch-mode messages.
+The CloudEvents within a batch are largely independent from one another: there
+is no restriction that they have the same source, producer, content type etc.
+There are just two restrictions, however:
+
+- All CloudEvents within the same batch MUST have the same `specversion`
+  attribute value.
+- All CloudEvents within the same batch MUST be distinct, i.e. the
+  (`source`, `id`) attribute value pair for each CloudEvent in the batch must
```
Comment (duglin) [inline cloudevents/spec.md:158]: s/must/MUST/

**Comment by duglin** [comment]
LGTM
thanks

File: cloudevents/spec.md (hunk 1 above)
Comment (jskeet) [reply cloudevents/spec.md]: Done, thanks.

File: cloudevents/spec.md [hunk 2]
```diff
@@ -148,6 +148,15 @@ as one that is unaware of CloudEvents.
 A "batch-mode message" is one where multiple (zero or more) events are
 encoded in a single message body, according to a specific event format. Not
 all event formats or protocol bindings support batch-mode messages.
+The CloudEvents within a batch are largely independent from one another: there
+is no restriction that they have the same source, producer, content type etc.
+There are just two restrictions, however:
+
+- All CloudEvents within the same batch MUST have the same `specversion`
+  attribute value.
+- All CloudEvents within the same batch MUST be distinct, i.e. the
```
Comment (duglin) [inline cloudevents/spec.md:157]: On the 3/2 call we agreed to remove the 2nd bullet

Comment (jskeet) [reply cloudevents/spec.md]: Done in extra commit.

**Comment by duglin** [comment]
LGTM

**Comment by duglin** [comment]
Approved on the 3/9 call

**Pull request merged**